codec/src/test/java/io/netty/handler/codec/DatagramPacketDecoderTest.java

```diff
@@ -56,4 +59,31 @@ public void testDecode() {
         assertTrue(channel.writeInbound(new DatagramPacket(content, recipient, sender)));
         assertEquals("netty", channel.readInbound());
     }
+
+    @Test
+    public void testNotSharable() {
+        DatagramPacketDecoder decoder = new DatagramPacketDecoder(new MessageToMessageDecoder<ByteBuf>() {
+            @Override
+            protected void decode(ChannelHandlerContext ctx, ByteBuf msg, List<Object> out) throws Exception {
+                // NOOP
+            }
+        });
+        assertFalse(decoder.isSharable());
```

Comment:
nit: should we assert that this value equals the value of the `MessageToMessageDecoder#isSharable` ?